The Great Input Refactor

EnnuiL opened the pull request:
This PR targets input-related mappings (specially those related to key binds, but not only them) and has the aim of making them way more readable than before.
The main changes were:
- `KeyBinding` becomes `KeyBind`, reducing verboseness and matching Vanilla's naming of the Key Binds screen
- `InputUtils` has been appropriately moved to the Blaze3D package
- The format of `InputUtils`' constants were edited in order to be more clear (From `GLFW_INPUT_X` to `INPUT_X_CODE`)
- The rest of `InputUtils`' constants are now filled with the help of Mojmap
- Expanded the unpicking of methods that uses key codes, now featuring mouse buttons, modifiers, and more
- Made the usage of `keyCode` and `code` for arguments containing key codes consistent (currently `code` -> `keyCode`)
- Refactored many input-related classes in order to get rid of inaccurate names and generally improve them
- ...and more!

I'd like to thank everyone who has given feedback during its draft, it has helped a lot with the PR

EnnuiL committed:
The Great Input Refactor (Part I)

EnnuiL committed:
Add back extra mouse buttons
i forgot ;-;

EnnuiL committed:
The Great Input Refactor (Part I.V)
TODO: figure out what's going on with the left mouse button

EnnuiL committed:
The Great Input Refactor (Part I.75)
roman numbers are getting boring

EnnuiL committed:
The Great Input Refactor (Part II)
the whole part thingy probably became meaningless

EnnuiL committed:
The Great Input Refactor (Part III Part I)
Mouse got refactored today!
Next: Keyboard and more

EnnuiL committed:
The Great Input Refactor (Part III Part II)
I think I'm ready!

EnnuiL committed:
The Great Input Refactor (Epilogue)
Turns out I forgot some bits!

github-actions[bot] committed:
Merge branch '22w05a' into 1.18.1

OroArmor commented on mappings/com/mojang/blaze3d/platform/InputUtil.mapping:
```diff
@@ -0,0 +1,197 @@
+CLASS net/minecraft/unmapped/C_vdmgqzcl com/mojang/blaze3d/platform/InputUtil
+	FIELD f_aomdviqh MODIFIER_CONTROL_CODE I
```
the modifiers are flags

OroArmor replied:
```suggestion
	FIELD f_aomdviqh MODIFIER_CONTROL_FLAG I
```
since this is the only one in the class, just apply the suggestion.

OroArmor commented on mappings/net/minecraft/client/Mouse.mapping:
```diff
@@ -33,30 +34,30 @@ CLASS net/minecraft/unmapped/C_zgebrxop net/minecraft/client/Mouse
 			COMMENT the window handle
-		ARG 3 horizontal
+		ARG 3 scrollDeltaX
 			COMMENT the horizontal scroll distance
```
```suggestion
			COMMENT the horizontal scroll delta
```

EnnuiL replied:
Huh, I swear that I deleted that particular javadoc; I thought that it was unnecessary with the new variable names

OroArmor commented on mappings/net/minecraft/client/Mouse.mapping:
```diff
@@ -33,30 +34,30 @@ CLASS net/minecraft/unmapped/C_zgebrxop net/minecraft/client/Mouse
 			COMMENT the horizontal scroll distance
-		ARG 5 vertical
+		ARG 5 scrollDeltaY
 			COMMENT the vertical scroll distance
```
```suggestion
			COMMENT the vertical scroll delta
```

OroArmor commented on mappings/net/minecraft/client/gui/screen/option/KeyBindsScreen.mapping:
```diff
@@ -1,11 +1,11 @@
+	COMMENT Represents the key binds screen.
 	COMMENT
-	COMMENT <p>It shows a list of the registered {@link KeyBinding} with buttons to bind or reset them.
+	COMMENT <p>It shows a list of the registered {@link KeyBind} with buttons to bind or reset them.
```
```suggestion
	COMMENT <p>It shows a list of the registered {@link KeyBind}s with buttons to bind or reset them.
```

OroArmor commented on mappings/net/minecraft/client/option/KeyBind.mapping:
```diff
@@ -1,37 +1,38 @@
 	FIELD f_isrvhtqu MOVEMENT_CATEGORY Ljava/lang/String;
 	FIELD f_kcphxfdy GAMEPLAY_CATEGORY Ljava/lang/String;
-	FIELD f_khpbnlde KEY_TO_BINDINGS Ljava/util/Map;
+	FIELD f_khpbnlde KEY_BINDS_BY_KEY Ljava/util/Map;
```
`KEY_BINDS_BY_KEY_CODE`? the last `KEY` is vauge

EnnuiL replied:
Key as in InputUtil's Key class, so, yeah, it's not only the key code

OroArmor reviewed (changes requested):
Pretty good. mostly nitpicks

EnnuiL committed:
Fix a flag's name

Co-authored-by: Eli Orona <eliorona@live.com>

EnnuiL committed:
Fix the lack of a plural

Co-authored-by: Eli Orona <eliorona@live.com>

EnnuiL committed:
Nuked Mouse's only javadoc
It made sense when the variables were named `horizontal` and `vertical`, but now, they are self-explanatory

github-actions[bot] committed:
Merge branch '22w06a' into 1.18.1

IotaBread commented on mappings/net/minecraft/client/gui/widget/option/KeyBindListWidget.mapping:
```diff
@@ -10,14 +10,14 @@ CLASS net/minecraft/unmapped/C_aarntxpx net/minecraft/client/gui/widget/option/K
-			ARG 2 binding
-			ARG 3 bindingName
+		FIELD f_myzuvtet key Lnet/minecraft/unmapped/C_sygikzty;
+		METHOD <init> KeyBindEntry (Lnet/minecraft/unmapped/C_aarntxpx;Lnet/minecraft/unmapped/C_sygikzty;Lnet/minecraft/unmapped/C_rdaqiwdt;)V
```
```suggestion
		METHOD <init> (Lnet/minecraft/unmapped/C_aarntxpx;Lnet/minecraft/unmapped/C_sygikzty;Lnet/minecraft/unmapped/C_rdaqiwdt;)V
```

EnnuiL replied:
Oh, I guess Enigma did that automatically

EnnuiL committed:
Clean up KeyBindEntry's constructor

Co-authored-by: ByMartrixX <bymartrixx@gmail.com>

EnnuiL committed:
Wait, did that last commit cause trouble?

This reverts commit badbc3b13570b5d07046259ba2fde943a6573f79.

EnnuiL committed:
Oh, it didn't

This reverts commit 9b2ef71ad8c91e17f873d372d0bf177b37b8d208.

IotaBread commented on mappings/net/minecraft/client/input/KeyboardInput.mapping:
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/unmapped/C_cxdvjlve net/minecraft/client/input/KeyboardInput
-	FIELD f_cgjjsojv settings Lnet/minecraft/unmapped/C_lscptbhn;
+	FIELD f_cgjjsojv options Lnet/minecraft/unmapped/C_lscptbhn;
+	FIELD f_isclmpwk SLOW_DOWN_FACTOR D
```
This field was removed, and it causes an NPE in the build action

EnnuiL replied:
Oh, I'll fix that

EnnuiL committed:
Cover new mappings, unpick categories
Credits to FoundationGames for helping with the new names

OroArmor committed:
Merge branch '22w06a' into 1.18.1
